pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
+	// Required: The IP of this endpoint.
+	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
+	IP string
+
+	// The ports exposed on this endpoint.
+	// FIXME: this would be nicer as a map of name -> struct
```

Comment:
If we think we want to change this, we should re-evaluate the overall convention. Maps would definitely make some things easier and/or nicer.